include/lsst/ctrl/events/Event.h

```diff
@@ -112,20 +112,22 @@ class Event
     virtual void populateHeader(cms::TextMessage* msg) const;
     void marshall(cms::TextMessage *msg);
 
-
-
 protected:
     PropertySet::Ptr _psp;
+    PropertySet::Ptr _filterable;
     set<std::string> _keywords;
     void _init();
-    void _constructor(const std::string& runid, const PropertySet& ps);
+    void _constructor(const std::string& runid, const PropertySet& ps, const PropertySet& filterable);
 
     template<typename T>void add(const std::string& name, const std::string& tag, const PropertySet& ps, boost::property_tree::ptree& child);
 
+
 private:
     std::string marshall(const PropertySet& ps);
     PropertySet::Ptr processTextMessage(cms::TextMessage *textMessage);
     PropertySet::Ptr unmarshall(const std::string& text);
+    PropertySet::Ptr parsePropertySet(boost::property_tree::ptree child);
+    bool addDataItem(std::string typeInfo, boost::property_tree::ptree& item, std::string key, PropertySet& ps);
 };
```

Comment:
Passing a lot of things by value here (e.g., `boost::property_tree::ptree child`, `std::string key`) rather than by reference.

Which of these can afford to be `const`?